NitroxModel/DataStructures/ThreadSafeDictionary.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using ProtoBufNet;
+
+namespace NitroxModel.DataStructures
+{
+    [DebuggerDisplay("Items = {" + nameof(Entries) + "}")]
+    [ProtoContract]
+    public class ThreadSafeDictionary<TKey, TValue> : IDictionary<TKey, TValue>
```

Comment:
Forgive my ignorance, im new to the source code of Nitrox, but would `ConcurrentDictionary` not be on option instead of re-implementing things?

Reply:
I remade it so that we have more control over serialization and how the data is actually processed internally. This allows us to optimize for Nitrox.

For general purpose applications I would have used ConcurrentDictionary.